src/components/container/LabResultsBloodType/LabResultsBloodType.tsx

```diff
@@ -0,0 +1,165 @@
+        var div: any, width: any = (getScrollbarWidth as any).width;
+        if (width === undefined) {
+            div = document.createElement('div');
+            div.innerHTML = '<div style="width:50px;height:50px;position:absolute;left:-50px;top:-50px;overflow:auto;"><div style="width:1px;height:100px;"></div></div>';
+            div = div.firstChild;
+            document.body.appendChild(div);
+            width = (getScrollbarWidth as any).width = div.offsetWidth - div.clientWidth;
+            document.body.removeChild(div);
+        }
+        return width;
+    };
+
+    if (!model) {
+        return <div className="mdhui-health-preview-section"><LoadingIndicator /></div>
+    }
+
+    if (model && (!model.BloodTypeLabs || model.BloodTypeLabs.length == 0)) {
+        return null;
+    }
+
+    // support any onClick function, or default to an english/spanish link
+    function drilldown() {
+        var linkTarget: string;
+        if (!model) { return; }
+        if (props.onClick) {
+            props.onClick();
+            return;
+        }
+        if (!props.externalInfoLink) { return; }
+        if (!props.externalInfoLinkDefault) { return; }
+        linkTarget = props.externalInfoLinkDefault;
+        if (MyDataHelps.language == "es" && props.externalInfoLinkSpanish) linkTarget = props.externalInfoLinkSpanish;
+        MyDataHelps.openEmbeddedUrl(linkTarget);
+    }
+
+    // Sometimes the data will only have either the type or the rH factor
+    // Combine those into a unified value, if the first value doesn't contain both
+    var summaryResults = function(bloodTypeLabs: any) {
+        var type = "";
+        var rH = "";
+        for (const bloodTypeLab of bloodTypeLabs) {
+            if (!bloodTypeLab?.MostRecentValue) continue;
+            var mostRecentValue = bloodTypeLab.MostRecentValue.toString().toLowerCase();
+            if (type == "") {
+                if (mostRecentValue == "a" || mostRecentValue.startsWith("a ")) {
+                    type = "A";
+                } else if (mostRecentValue == "ab" || mostRecentValue.startsWith("ab ")) {
+                    type = "AB";
+                } else if (mostRecentValue == "b" || mostRecentValue.startsWith("b ")) {
+                    type = "B";
+                } else if (mostRecentValue == "o" || mostRecentValue.startsWith("o ")) {
+                    type = "O";
+                }
+            }
+            if (rH == "") {
+                if (mostRecentValue.includes("pos")) {
+                    rH = "Positive";
+                } else if (mostRecentValue.includes("neg")) {
+                    rH = "Negative";
+                }
+            }
+        }
+        return type + " " + rH;
+    }
+
+    return <Action
+        bottomBorder
+        title={ language ("blood-type") }
+        titleIcon={<img className="mdhui-health-preview-icon" src={ iconBloodType} alt="Blood Type" />}
+        onClick={ drilldown }
```

Comment:
Can the model be checked for here vs in drilldown?

Reply:
Updated